File: sdks/typescript/examples/list_products.ts
import { Puxbay, PuxbayConfig } from '../src';

async function main() {
    const apiKey = process.env.PUXBAY_API_KEY;
    if (!apiKey) {
        console.error('PUXBAY_API_KEY must be set');
        process.exit(1);
    }

    const config = new PuxbayConfig({ apiKey });
    const client = new Puxbay(config);

    console.log('Fetching products...');
    try {
        const products = await client.products.list(1);
        products.results.forEach(product => {
            console.log(`- ${product.name} ($${product.price})`);
        });
    } catch (error) {
        console.error('Error:', error);
    }
}

main();

File: sdks/rust/examples/list_products.rs
use puxbay_sdk::{Puxbay, PuxbayConfig};
use std::error::Error;
use std::env;

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    let api_key = env::var("PUXBAY_API_KEY").expect("PUXBAY_API_KEY must be set");
    
    let config = PuxbayConfig::builder()
        .api_key(&api_key)
        .build();

    let client = Puxbay::new(config);

    println!("Fetching products...");
    let products = client.products().list(1).await?;

    for product in products.results {
        println!("- {} (${})", product.name, product.price);
    }
    
    Ok(())
}
